packages/office-ui-fabric-react/src/utilities/keytips/KeytipManager.ts

```diff
@@ -0,0 +1,430 @@
+          }
+
+          // Reset currentSequence
+          this.currentSequence = '';
+          // Return pointer to its parent
+          this.keytipTree.currentKeytip = this.keytipTree.getNode(currKtp.parent);
+          // Show children keytips of the new currentKeytip
+          this.showKeytips(this.keytipTree.getChildren());
+        }
+      }
+    } else if (transitionKeysContain(this._enableSequences, transitionKey) && !currKtp) {
+      // If key sequence is in 'entry sequences' and currentKeytip is null, we enter keytip mode
+      this.enterKeytipMode();
+    }
+  }
+
+  /**
+   * Processes inputs from the document listener and traverse the keytip tree
+   *
+   * @param key - Key pressed by the user
+   */
+  public processInput(key: string): void {
+    // Concat the input key with the current sequence
+    const currSequence: IKeySequence = this.currentSequence + key;
+    let currKtp = this.keytipTree.currentKeytip;
+
+    // currentKeytip must be defined, otherwise we haven't entered keytip mode yet
+    if (currKtp) {
+      const node = this.keytipTree.getExactMatchedNode(currSequence, currKtp);
+      if (node) {
+        this.keytipTree.currentKeytip = currKtp = node;
+        const currKtpChildren = this.keytipTree.getChildren();
+
+        // Execute this node's onExecute if defined
+        if (currKtp.onExecute) {
+          currKtp.onExecute(this._getKeytipDOMElement(currKtp.id));
+          // Reset currKtp, this might have changed from the onExecute
+          currKtp = this.keytipTree.currentKeytip;
+        }
+
+        // To exit keytipMode after executing keytip we should check if currentKeytip has no children and
+        // if the node doesn't have children nodes
+        if (currKtpChildren.length === 0 && !currKtp.hasChildrenNodes) {
+          this.exitKeytipMode();
+        } else {
+          // Show all children keytips
+          this.showKeytips(currKtpChildren);
+        }
+
+        // Clear currentSequence
+        this.currentSequence = '';
+        return;
+      }
+
+      const partialNodes = this.keytipTree.getPartiallyMatchedNodes(currSequence, currKtp);
+      if (partialNodes.length > 0) {
+        // We found nodes that partially match the sequence, so we show only those
+        // Omit showing persisted nodes here
+        const ids = partialNodes.filter((partialNode: IKeytipTreeNode) => {
+          return !partialNode.persisted;
+        }).map((partialNode: IKeytipTreeNode) => { return partialNode.id; });
+        this.showKeytips(ids);
+        // Save currentSequence
+        this.currentSequence = currSequence;
+      }
+    }
+  }
+
+  /**
+   * Gets all IKeytipProps from this.keytips
+   *
+   * @returns {IKeytipProps[]} All keytips stored in the manager
+   */
+  public getKeytips(): IKeytipProps[] {
+    return this.keytips.map((uniqueKeytip: IUniqueKeytip) => {
+      return uniqueKeytip.keytip;
+    });
+  }
+
+  private _addKeytipToQueue(keytipID: string) {
```

Comment:
This is a new development, I kept it in to help with discussion. Here's the scenario: you trigger a button keytip that has a menu, and that menu has keytips as well. That DOM may go through multiple mounts and unmounts, corresponding with multiple keytip registers and unregisters. We don't know exactly when that DOM will do its last "mount" so this timeout will keep being set until we don't register a keytip for 500ms (to possibly be changed to be smaller), and then we will show the keytips. This allows the keytip to be correctly positioned on its element